addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,138 @@
+                                                "X-Rewrite-URL: "+PATH, "X-Original-URL: "+PATH,"Referer: "+PATH, "X-Custom-IP-Authorization: 127.0.0.1","X-Originating-IP: 127.0.0.1","X-Forwarded-For: 127.0.0.1","X-Remote-IP: 127.0.0.1","X-Client-IP: 127.0.0.1","X-Host: 127.0.0.1","X-Forwared-Host: 127.0.0.1"
+                                        }));
+
+                //sending payloads in URL.
+                for(String i:upayloads){
+                    String TMP_URI = SCHEMA+"://"+HOST+""+i;
+                    HttpMessage req_with_payload = new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI,true));
+                    sendAndReceive(req_with_payload);
+                    if(req_with_payload.getResponseHeader().getStatusCode() == HttpStatusCode.OK){
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(Constant.messages.getString("ascanalpha.forbiddenBypass.name"))
+                                .setDescription(Constant.messages.getString("ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(TMP_URI)
+                                .setMessage(req_with_payload)
+                                .raise();
+                    }
+
+                }
+
+                //sending payloads in Header.
+                for(String h:hpayloads){
+                    String TMP_URI = SCHEMA+"://"+HOST+PATH;
+                    if(h.contains("X-Rewrite-URL")){
+                        TMP_URI = SCHEMA+"://"+HOST+"/anything";
+                    }else if(h.contains("X-Original-URL")){
+                        TMP_URI = SCHEMA+"://"+HOST+"/";
+                    }else if(h.contains("Referer")){
+                        TMP_URI = SCHEMA+"://"+HOST+"/anything";
+                    }
+                    HttpMessage req_with_payload1 = new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI,true));
+                    req_with_payload1.getRequestHeader().setHeader(h.split(":")[0], h.split(":")[1]);
+                    sendAndReceive(req_with_payload1);
+                    if(req_with_payload1.getResponseHeader().getStatusCode() == HttpStatusCode.OK){
+                        System.out.println(req_with_payload1.getResponseHeader().getStatusCode());
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(Constant.messages.getString("ascanalpha.forbiddenBypass.name"))
+                                .setDescription(Constant.messages.getString("ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(FULL_URL)
+                                .setMessage(req_with_payload1)
+                                .raise();
+                    }
+                }
+
+
+
+            }
+        } catch (IOException | URISyntaxException e) {
+            e.printStackTrace();
```

Comment:
Please use a proper logger, you can look at other scan rule classes for examples.